src/mbgl/annotation/annotation_tile.cpp

```diff
@@ -19,37 +19,105 @@ AnnotationTile::~AnnotationTile() {
     annotationManager.removeTile(*this);
 }
 
-void AnnotationTile::setNecessity(Necessity) {}
+void AnnotationTile::setNecessity(Necessity) {
+}
+
+class AnnotationTileFeatureData {
+public:
+    AnnotationTileFeatureData(const AnnotationID id_,
+                              FeatureType type_,
+                              GeometryCollection geometries_,
```

Comment:
So do you think it should be `GeometryCollection &&` to avoid the extra copy? If so, I presume that would require a separate refactor?